woodwork/utils.py

```diff
@@ -23,6 +23,41 @@ def import_or_none(library):
         return None
 
 
+def import_or_raise(library, error_msg):
+    '''
+    Attempts to import the requested library.  If the import fails, raises an
+    ImportError with the supplied error message.
+
+    Args:
+        library (str): the name of the library
+        error_msg (str): error message to return if the import fails
+    '''
+    try:
+        return importlib.import_module(library)
+    except ImportError:
+        raise ImportError(error_msg)
+
+
+def import_and_configure_koalas(compute_ops_on_diff_frames=True):
+    """
+    Import the koalas library. Set the compute.ops_on_diff_frames as an optimization.
+
+    Args:
+        compute_ops_on_diff_frames (bool): the koalas compute.ops_on_diff_frames argum,ent
+
+    Returns:
+        A reference to the koalas module
+    """
+    ks = import_or_none('databricks.koalas')
+    if ks is None:
+        return None
+    try:
+        ks.set_option('compute.ops_on_diff_frames', compute_ops_on_diff_frames)
```

Comment:
Something which we could discuss separately is making this configurable.

I would categorize this "compute.ops_on_diff_frames" option as global state, meaning something users could set/unset independent of woodwork. Ideally, low-level libraries like woodwork and evalml would either a) not change global state or b) have an opinion on changing global state but give users a way to override.

My suggestion on how to do that would be to add a config for this option. I didn't do that in this PR as that's not the immediate issue, but that's come up in discussions with @boopboopbeepboop recently. Could be a good separate issue for woodwork. We have a similar issue to solve in evalml RE configuring the logger.

@rwedge FYI

Reply:
Checked into this - we need to have this option set to be able to reassign new columns to a dataframe when a type change happens. In this case we create a new series with the appropriate type and replace the existing series with the new series. So without this option set, we can't do much with Woodwork and Koalas dataframes.

We can handle this more gracefully though, and avoid an error when the user isn't trying to use Koalas even though they might have it installed.